glib/src/gstring.rs

```diff
@@ -404,6 +404,48 @@ impl<'a> ToGlibPtr<'a, *mut c_char> for GString {
     }
 }
 
+#[doc(hidden)]
+impl<'a> FromGlibContainer<*const c_char, *const i8> for GString {
+    unsafe fn from_glib_none_num(ptr: *const i8, num: usize) -> Self {
+        if num == 0 {
+            return Self::from("");
+        }
+        let mut bytes = Vec::with_capacity(num);
+
+        for i in 0..num {
+            let b = std::ptr::read(ptr.add(i)) as u8;
+            bytes.push(b);
+        }
+        CStr::from_bytes_with_nul_unchecked(bytes.as_slice()).into()
```

Comment:
Does this actually work? I believe that function requires the vec to be NUL terminated or otherwise it's UB. You could add a 0 byte in the end though

Reply:
Should be working now